librz/type/base.c

```diff
@@ -199,6 +203,8 @@ static RzBaseType *get_union_type(RzTypeDB *typedb, const char *sname) {
 	}
 	free(sdb_members);
 
+	base_type->size = base_type->union_data.members.len * base_type->union_data.members.elem_size;
```

Comment:
Union size is also works differently, it's something like `RZ_MAX()`. See https://github.com/rizinorg/rizin/pull/1047/files#diff-f936f8d2d463f6bcab3c0c7bf452143da7ae6542ddb3bee7ae2e281cb948918dR660